### PR TITLE
Fix bugs with reinforcing floors

### PR DESCRIPTION
**Pull request by FlameArrow57**
[ACTIONS][BUG]

## About the PR 
This PR fixes bugs with reinforcing floors by reworking it into an action bar.

Fixes a bug where you could click on a floor multiple times and it would reinforce the floor more than one time.

Fixes a bug where you could start reinforcing a floor, step away, then move back to your original location to complete the reinforce.

## Why's this needed? 
Bug fixes

Action bar would be nice

### CODE REVIEW

File: code/turf/floors.dm [hunk 1]
```diff
@@ -1848,20 +1848,10 @@ DEFINE_FLOORS(solidcolor/black/fullbright,
...
-					if (C)
-						C.change_stack_amount(-2)
-						if (C:amount <= 0)
-							qdel(C) //wtf
-
-						if (C.material)
-							src.setMaterial(C.material)
-
-					playsound(src, "sound/items/Deconstruct.ogg", 80, 1)
+				SETUP_GENERIC_PRIVATE_ACTIONBAR(user, src, 3 SECONDS, /turf/simulated/floor/proc/reinforce, C, C.icon, C.icon_state, null, INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_ATTACKED | INTERRUPT_STUNNED | INTERRUPT_ACTION)
```
Comment (ZeWaka) [inline code/turf/floors.dm:1854]: Why would this be private

Comment (FlameArrow57) [reply code/turf/floors.dm]: Was just for theme of how it was before, changed

**Comment by ZWRh3** [comment]
doesnt this make reinforced floor construction **9** times slower?

**Comment by FlameArrow57** [comment]
I don't think so? 3 seconds is used still

**Pull request merged**